shor/gates.py

```diff
@@ -22,10 +22,12 @@ class _Gate(_Layer):
     def symbol(self):
         return self.__class__.__name__.lower()
 
-    def __init__(self, *qbits: QbitOrIterable, **kwargs):
+    def __init__(self, bool=None, cbit=None, *qbits: QbitOrIterable, **kwargs):
```

Comment:
These arguments are in the wrong order.

For python arguments look like this

For class methods, they always start with self.
```
class_method(self, pos_arg1, pos_arg2, *args, key_arg1='abc', key_arg2=None, **kwargs)
```
Here, you have two optional key word arguments (bool, and cbit) in front of the positional args capture  `*qbits`

Also, `bool` is a reserved word in python https://realpython.com/lessons/reserved-keywords/

So you don't want to use that as a variable name.

What was the intention of the bool?

Reply:
I would make these changes elsewhere.

Gates should be purely functional operations (no measurements, only quantum)

`_Operation` would be a better place to have optional qbits.

Reply:
Intention of bool is to allow users to say for example apply H gate to Qubit 2 if cbit 0 is 1

Reply:
Okay Ill change conditional to an operation